stat_var_renaming/stat_var_renaming.py

```diff
@@ -0,0 +1,631 @@
+  Args:
+    Row -> A preprocessed dataframe row with dependent variables removed and
+      all constraints, values, and populations remove.
+
+  Returns:
+    String mapping between the provided row to the corresponding human readable
+    dcid format: <?statType>_<mProp>_<popType>_<v1>_<v2>_..._<mQual>_<mDenom>
+  """
+  # Add measured property and population type. e.g. Count_InsuranceClaim.
+  human_string = (f"{capitalizeFirst(row['measuredProp'])}" +
+                  f"_{capitalizeFirst(row['populationType'])}")
+
+  # StatType (e.g. median) is prepending if the stat type is not measuredValue.
+  stat_type = row['statType']
+  if stat_type != "measuredValue" and stat_type != "Unknown":
+    human_string = (capitalizeFirst(stat_type.replace("Value", ""))
+                    + "_" + human_string)
+
+  # Postpend renamed constraint fields.
+  row_constraints = min(row['numConstraints'], MAX_CONSTRAINTS_WITH_DPV)
+  for num in range(1, row_constraints + 1): 
+    if not pd.isna(row[f"v{num}"]):
+      human_string += "_" + row[f"v{num}"]
+
+  # Postpend nmeasurement qualifier if it exists.
+  measurement_qualifier = row['measurementQualifier']
+  if not pd.isna(measurement_qualifier):
+    human_string = f"{human_string}_{measurement_qualifier}" 
+
+  # Postpend measurement denominator if it exists.
+  measurement_denominator = row['measurementDenominator']
+  if not pd.isna(measurement_denominator):
+    # Special Case: PerCapita which is directly appended.
+    if measurement_denominator == "PerCapita":
+      human_string = f"{human_string}_PerCapita"
+    # MDoms that are properties (all lower case) are added as Per(Mdom).
+    elif measurement_denominator[0].islower():
+      human_string = (
+        f"{human_string}_Per{capitalizeFirst(measurement_denominator)}")
+    # Everything else is AsAFractionOf.
+    else:
+      human_string = f"{human_string}_AsAFractionOf{measurement_denominator}"
+  return human_string
+
+def ensure_no_overlapping_stat_vars(stat_vars):
+  """ Ensures no collisions between statistical variables.
+  
+  This function ensures that there are not two distinct statistical variables
+  (e.g. different constraint, values, ect) that have the same name. There may
+  be two rows in the dataframe with the same name that have differences that
+  do not result in different statistical variables. E.g. same properties
+  except for measurement method.
+
+  Args:
+    stat_vars -> Dataframe of stat_vars with human readable name generated.
+
+  Returns:
+    stat_vars dataframe with no duplicate rows by name.
+  
+  Raises:
+    Assertion error if two distinct statistical variables evaluate to the same
+    name.
+  """
+  # Consider the subset of statistical variables that have at least one overlap.
+  overlapping_stat_var_names = (
+    stat_vars[stat_vars.duplicated('HumanReadableName', keep=False)])
+
+  # Check to make sure that none of these are actually different stat vars.
+  properties_of_statistical_variables = 
```

Comment:
@IanCostello  can you add a TODO(tjann): update this list--measurementDenominator and scalingFactor are no longer SV props.

Reply:
Added